### PR TITLE
Add basic functions to sample from audio and video files

### PR DESCRIPTION
**Pull request by hohsiangwu**
This is my attempt to implement sampling video and audio using pescador.

I tested it locally with few files, I am not sure about the performance while sampling the real directory on the server. That would be my next test to run.

You can do this locally by loading in the code and do.
```
gen = data_generator('data/eval_segments')

for batch in gen:
    train(batch)
```

Feedbacks are welcome and I will report back how it works with real data on the server.

### CODE REVIEW

**Comment by jtcramer** [comment]
I forgot, but make sure to set the random seed in `data_generator()` for reproducibility

**Comment by hohsiangwu** [comment]
Updated based on comments and test with real `eval_segments` data on the server.

Be aware that you need to use the system sox so it has flac support, the one comes with module install does not work.

File: l3embedding/train.py [hunk 1]
```diff
@@ -1,51 +1,142 @@
...
+
+    video_data = skvideo.io.vread(video_file)
+    audio_file = video_to_audio(video_file)
+
+    if random.random() < 0.5:
+        audio_file = random.choice([af for af in audio_files if af != audio_file])
+        label = 0
+    else:
+        label = 1
+
```
Comment (jtcramer) [inline l3embedding/train.py]: Audio should be loaded here, so that it's not opened every time we sample

File: l3embedding/train.py [hunk 2]
```diff
@@ -1,51 +1,142 @@
...
+            'audio': sample_audio_data[:,0],
+            'label': label,
+            'audio_file': audio_file,
+            'video_file': video_file,
+            'audio_start': audio_start,
+            'video_start': video_start
+        }
+
+
+def data_generator(data_dir, k=32, batch_size=64):
```
Comment (jtcramer) [inline l3embedding/train.py]: Should take in random seed eventually

**Pull request merged**